src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java

```diff
@@ -378,40 +386,64 @@ public void onOpened()
         createXpBar();
         createSkillContent();
 
-        resourceList.setDataProvider(() -> getOpenRequestsOfCitizen().size(), (int index, Pane rowPane) ->
-        {
-            @SuppressWarnings(RAWTYPES)
-            final ImmutableList<IRequest> openRequests = getOpenRequestsOfCitizen();
-            if (index < 0 || index >= openRequests.size())
-            {
-                return;
-            }
+        resourceList.setDataProvider(new ScrollingList.DataProvider() {
 
-            final IRequest<?> request = openRequests.get(index);
-            final ItemIcon exampleStackDisplay = rowPane.findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_STACK, ItemIcon.class);
-            final List<ItemStack> displayStacks = request.getDisplayStacks();
+            private List<RequestWrapper> requestWrappers = null;
 
-            if (!displayStacks.isEmpty())
+            @Override
+            public int getElementCount()
             {
-                exampleStackDisplay.setItem(displayStacks.get((lifeCount / LIFE_COUNT_DIVIDER) % displayStacks.size()));
+                requestWrappers = getOpenRequestTreeOfCitizen();
+                return requestWrappers.size();
             }
-            else
+
+            @Override
+            public void updateElement(final int index, final Pane rowPane)
             {
-                final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
-                logo.setVisible(true);
-                logo.setImage(request.getDisplayIcon());
-            }
+                if (index < 0 || index >= requestWrappers.size())
+                {
+                    return;
+                }
+
+                final RequestWrapper wrapper = requestWrappers.get(index);
+                final Box wrapperBox = rowPane.findPaneOfTypeByID(WINDOW_ID_REQUEST_BOX, Box.class);
+                wrapperBox.setPosition(wrapperBox.getX() + 10*wrapper.getDepth(), wrapperBox.getY());
+                wrapperBox.setSize((int) (wrapperBox.getParent().getWidth() * 0.9f - 10*wrapper.getDepth()), wrapperBox.getHeight());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.9f to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a "double" or "BigDecimal" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS818)